test/mocks/pinecone.py

```diff
@@ -58,10 +57,23 @@ def upsert(self, vectors: List[tuple], namespace: str = ""):
             upsert_count += 1
         return {"upserted_count": upsert_count}
 
-    def describe_index_stats(self):
+    def update(self, namespace: str, id: str, set_metadata: dict):
+        # Get existing item metadata
+        meta = self.index_config.namespaces[namespace][id]["metadata"]
+        # Add new metadata to existing item metadata
+        self.index_config.namespaces[namespace][id]["metadata"] = {**meta, **set_metadata}
```

Comment:
This code is clear, so feel free to keep. Just adding an option in case you like it (I'm personally fond of set and dict operators, but not everybody is :smile:):

```suggestion
        # Merge new metadata with existing item metadata (set_metadata takes priority in case of key collisions)
        self.index_config.namespaces[namespace][id]["metadata"] |= set_metadata
```

Reply:
I think these operators are great, and I'd love to use them, but they weren't they introduced in Python 3.9? And Haystack allows >=3.7 - isn't this an issue?

Reply:
:facepalm: Sorry! You're totally right :facepalm: